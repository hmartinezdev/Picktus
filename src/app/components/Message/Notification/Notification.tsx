import color from '@constants/colors';
import React, { Component } from 'react';

export interface INotificationProps {
  text: string;
  level: PicktusMessageDisplayType;
}

class Notification extends Component<INotificationProps> {
  public render(): React.ReactElement<Notification> {
    const { text, level } = this.props;
    return (
      <div>
        <div className={`notification notification--${level}`}>{text}</div>
        <style jsx>{`
          .notification {
            width: 200px;
            min-height: 100px;
            background-color: ${color.secondary};
            border-radius: 3px;
          }
        `}</style>
      </div>
    );
  }
}

export default Notification;
