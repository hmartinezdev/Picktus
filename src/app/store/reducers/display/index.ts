import display from './display-reducer';
export { IDisplayState } from './display.type';

export default display;
