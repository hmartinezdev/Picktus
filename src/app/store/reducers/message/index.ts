import message from './message-reducer';
export { IMessageState } from './message.type';

export default message;
