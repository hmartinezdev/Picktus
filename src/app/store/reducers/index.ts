import { combineReducers } from 'redux';
import display, { IDisplayState } from './display';
import message, { IMessageState } from './message';
import user, { IUserState } from './user';

export interface IReduxState {
  user: IUserState;
  display: IDisplayState;
  message: IMessageState;
}

export default combineReducers({
  display,
  message,
  user,
});
