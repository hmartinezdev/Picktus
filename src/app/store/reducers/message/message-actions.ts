import { ThunkAction } from 'redux-thunk';
import { IMessageState } from './message.type';

export type ThunkResult<R> = ThunkAction<R, IMessageState, undefined, DismissNotification>;

export enum TypeKeys {
  MESSAGE_DISMISS_NOTIFICATION = 'MESSAGE_DISMISS_NOTIFICATION',
  MESSAGE_DISMISS_SNACKBAR = 'MESSAGE_DISMISS_SNACKBAR',
}

export interface DismissNotification {
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION;
}

export const dismissNotification = (): DismissNotification => ({
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION,
});

export interface DismissSnackBar {
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR;
}

export const dismissSnackBar = (): DismissSnackBar => ({
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR,
});

export type MessageActions = DismissNotification | DismissSnackBar;
