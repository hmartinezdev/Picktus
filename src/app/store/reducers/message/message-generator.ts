import uuid from 'uuid';
import { IPicktusError, IPicktusMessage, PicktusMessageDisplay, PicktusMessageLevel } from './message.type';

export default function(
  text: string,
  level: PicktusMessageLevel,
  display: PicktusMessageDisplay
): IPicktusError | IPicktusMessage {
  return {
    display,
    id: uuid.v4(),
    level,
    text,
  };
}
