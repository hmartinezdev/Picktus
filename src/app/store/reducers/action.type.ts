import { DisplayActions } from './display/display-actions';
import { MessageActions } from './message/message-actions';
import { UserActions } from './user/user-actions';

enum TypeKeys {
  OTHER_ACTION = '__any_other_action_type__',
}

interface OtherAction {
  type: TypeKeys.OTHER_ACTION;
}

export type ActionTypes = UserActions | DisplayActions | MessageActions | OtherAction;
