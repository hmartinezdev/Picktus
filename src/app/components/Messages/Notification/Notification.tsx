import colors from '@constants/colors';
import { PicktusMessageLevel } from '@store/reducers/message';
import React, { Component } from 'react';

export interface INotificationProps {
  text: string;
  level: PicktusMessageLevel;
}

class Notification extends Component<INotificationProps> {
  public render(): React.ReactElement<Notification> {
    const { text, level } = this.props;
    return (
      <div>
        <div className={`notification notification--${level}`}>{text}</div>
        <style jsx>{`
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.2rem;
            font-family: 'Josefin Sans', sans-serif;
            color: ${colors.primary};
            width: 250px;
            min-height: 5rem;
            background-color: ${colors.secondary};
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--${PicktusMessageLevel.ERROR} {
            background-color: ${colors.error};
            color: ${colors.white};
          }

          .notification--${PicktusMessageLevel.SUCCESS} {
            background-color: ${colors.green};
            color: ${colors.white};
          }

          .notification--${PicktusMessageLevel.WARNING} {
            background-color: ${colors.warning};
            color: ${colors.primary};
          }
        `}</style>
      </div>
    );
  }
}

export default Notification;
