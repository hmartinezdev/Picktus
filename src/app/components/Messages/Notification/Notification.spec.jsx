import { shallow } from 'enzyme';
import React from 'react';
import Notification from './Notification';
import { PicktusMessageLevel } from '@store/reducers/message';

let wrapper;

const setup = (props = {}) => shallow(<Notification {...props} />);

describe('<Notification />', () => {
  beforeEach(() => {
    wrapper = setup();
  });

  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with different icon', () => {
    wrapper = setup({ level: PicktusMessageLevel.ERROR, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.SUCCESS, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.WARNING, text: 'error' });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.INFO, text: 'error' });
    expect(wrapper).toMatchSnapshot();
  });
});
