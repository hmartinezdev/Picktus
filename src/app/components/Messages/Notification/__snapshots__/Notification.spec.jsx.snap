// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render properly 1`] = `
<div>
  <div
    className="notification notification--undefined"
  />
  <style
    jsx={true}
  >
    
          .notification {
            display: flex;
            align-items: center;
            box-sizing: border-box;
            padding: 1rem;
            font-size: 1.2rem;
            font-family: 'Josefin Sans', sans-serif;
            color: #2d3047;
            width: 250px;
            min-height: 5rem;
            background-color: #FAC9B8;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }

          .notification--ERROR {
            background-color: #982649;
            color: #F8F8F7;
          }

          .notification--SUCCESS {
            background-color: #75ab88;
            color: #F8F8F7;
          }

          .notification--WARNING {
            background-color: #F6BD60;
            color: #2d3047;
          }
        
  </style>
</div>
`;
